src/crypto/index.js

```diff
@@ -246,8 +246,12 @@ Crypto.prototype.uploadDeviceKeys = function() {
     });
 };
 
+Crypto.prototype.updateCurrentKeyCount = function(currentKeyCount) {
+     _maybeUploadOneTimeKeys(this, currentKeyCount);
```

Comment:
we can't just call this here, because we don't want to start uploading one-time keys until we've caught up with the device messages - see https://github.com/krombel/matrix-js-sdk/blob/fa2982a1bf80fc5ae8994a7d9bb05faf1fcd54ba/src/crypto/index.js#L927.